specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-08-01-preview/costmanagement.json

```diff
@@ -0,0 +1,2877 @@
+        },
+        "metric": {
+          "description": "Metric to use when displaying costs.",
+          "type": "string",
+          "enum": [
+            "ActualCost",
+            "AmortizedCost",
+            "AHUB"
+          ],
+          "x-ms-enum": {
+            "name": "MetricType",
+            "modelAsString": true
+          }
+        },
+        "kpis": {
+          "description": "List of KPIs to show in Cost Analysis UI.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KpiProperties"
+          }
+        },
+        "pivots": {
+          "description": "Configuration of 3 sub-views in the Cost Analysis UI.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PivotProperties"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        }
+      }
+    },
+    "KpiProperties": {
+      "description": "Each KPI must contain a 'type' and 'enabled' key.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "KPI type (Forecast, Budget).",
+          "type": "string",
+          "enum": [
+            "Forecast",
+            "Budget"
+          ],
+          "x-ms-enum": {
+            "name": "KpiTypeType",
+            "modelAsString": true
+          }
+        },
+        "id": {
+          "description": "ID of resource related to metric (budget).",
+          "type": "string"
+        },
+        "enabled": {
+          "description": "show the KPI in the UI?",
+          "type": "boolean"
+        }
+      }
+    },
+    "PivotProperties": {
+      "description": "Each pivot must contain a 'type' and 'name'.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Data type to show in view.",
+          "type": "string",
+          "enum": [
+            "Dimension",
+            "TagKey"
+          ],
+          "x-ms-enum": {
+            "name": "PivotTypeType",
+            "modelAsString": true
+          }
+        },
+        "name": {
+          "description": "Data field to show in view.",
+          "type": "string"
+        }
+      }
+    },
+    "Resource": {
+      "description": "The Resource model definition.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource Id."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource name."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        },
+        "location": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Location of the resource."
+        },
+        "sku": {
+          "readOnly": true,
+          "type": "string",
+          "description": "SKU of the resource."
+        },
```

Comment:
this is pretty clearly actually defining some kind of TrackedResource or ProxyResource or Resource with allowed extra fields, so it should be defined as an "allOf" inherit from the common-types/v2 or v3, or you should just not use this, and reference e.g. TrackedResoruce directly.